### PR TITLE
chore(parameters.py): make script safer

### PR DESCRIPTION
**Pull request by sabinem**
as this script is only used and called from external it seems necessary to have tests for its external usage

functions that are only called from internal are marked as private

the replacing has been improved by adding NotFound values where replacement values were not found: that way with the zip method that is used here for the replacement, all existing values will be replaced correctly and it will be also understandable from the output file which values are missing

### CODE REVIEW

File: parameters.py [hunk 1]
```diff
@@ -3,80 +3,95 @@
...
+            placeholderValue.append("NotFound")
 
     # Replacing template
-    template = readTemplate(filepath)
-    templateFilled = replaceListParameters(template, placeholders, placeholderValue)
-    _ = saveConfigFile(templateFilled, outputfilepath)
+    template = __readTemplate(filepath)
+    templateFilled = __replaceListParameters(template, placeholders, placeholderValue)
+    outputfilepath = __saveConfigFile(templateFilled, outputfilepath)
+    print(f"Outputfile available: {outputfilepath}.")
```
Comment (caviri) [inline parameters.py]: Shouldn't be `f"Output file`...?

File: parameters.py [hunk 2]
```diff
@@ -3,80 +3,95 @@
...
+              f"variables to resolve the errors.")
 
-    print("PREPARATION OF CONFIG DONE")
 
+if __name__ == "__main__":
+    # This python script is called from external: it reads a config file and
+    # output file path: it replaces all placeholder values marked by {{ .. }}
+    # with environment variables of the same name: so {{ HELLO }} is replaced
+    # by the value in $HELLO
+    print("PREPARATION OF CONFIG FILE")
```
Comment (caviri) [inline parameters.py:90]: Maybe we can use here `logging.info`?

File: parameters.py [hunk 3]
```diff
@@ -3,80 +3,95 @@
...
     return content
 
-def replaceListParameters(content, listPlaceholders, parameterValue):
 
+def __replaceListParameters(content, listPlaceholders, parameterValue):
+    """
+    :param content:
+    :param listPlaceholders:
+    :param parameterValue:
+    :return:
```
Comment (caviri) [inline parameters.py]: Return `content`

**Review by caviri — APPROVED** [review]

File: parameters.py (hunk 2 above)
Comment (sabinem) [reply parameters.py:90]: @caviri Logging needs to be configured to work. This can be done. But for now I would like to keep it simple.

**Pull request merged**